inc/cablestrand.class.php

```diff
@@ -0,0 +1,196 @@
+
+      if (!$withtemplate) {
+         switch ($item->getType()) {
+            case __CLASS__ :
+               $ong    = [];
+               $ong[1] = _n('Item', 'Items', Session::getPluralNumber());
+               return $ong;
+         }
+      }
+      return '';
+   }
+
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $withtemplate = 0) {
+
+      if ($item->getType() == __CLASS__) {
+         switch ($tabnum) {
+            case 1 :
+               $item->showItems();
+               break;
+         }
+      }
+      return true;
+   }
+
+      /**
+    * Print the HTML array of items for a location
+    *
+    * @since 0.85
+    *
+    * @return void
+   **/
+   function showItems() {
+      global $DB;
+
+      $cablestrands_id = $this->fields['id'];
+
+      if (!$this->can($cablestrands_id, READ)) {
+         return false;
+      }
+
+      $queries = [];
+      $item = new Cable();
+
+      $table = getTableForItemType($item->getType());
+      $itemtype_criteria = [
+         'SELECT' => [
+            "$table.id",
+            new \QueryExpression($DB->quoteValue($item->getType()) . ' AS ' . $DB->quoteName('type')),
+         ],
+         'FROM'   => $table,
+         'WHERE'  => [
+            "$table.cablestrands_id"   => $cablestrands_id,
+         ]
+      ];
+      if ($item->maybeDeleted()) {
+         $itemtype_criteria['WHERE']['is_deleted'] = 0;
+      }
+      $queries[] = $itemtype_criteria;
+
+      $criteria = count($queries) === 1 ? $queries[0] : ['FROM' => new \QueryUnion($queries)];
+      $start  = (isset($_REQUEST['start']) ? intval($_REQUEST['start']) : 0);
+      $criteria['START'] = $start;
+      $criteria['LIMIT'] = $_SESSION['glpilist_limit'];
+
+      $iterator = $DB->request($criteria);
+
+      // Execute a second request to get the total number of rows
+      unset($criteria['SELECT']);
+      unset($criteria['START']);
+      unset($criteria['LIMIT']);
+
+      $criteria['COUNT'] = 'total';
+      $number = $DB->request($criteria)->next()['total'];
+
+      if ($number) {
+         echo "<div class='spaced'>";
+         Html::printAjaxPager('', $start, $number);
+
+         echo "<table class='tab_cadre_fixe'>";
+         echo "<tr><th>"._n('Type', 'Types', 1)."</th>";
+         echo "<th>".Entity::getTypeName(1)."</th>";
+         echo "<th>".__('Name')."</th>";
+         echo "<th>".__('Inventory number')."</th>";
+         echo "<th>"._n('Associated item', 'Associated items', 0)." (".__('Front').")"."</th>";
+         echo "<th>".Socket::getTypeName(1)." (".__('Front').")"."</th>";
+         echo "<th>"._n('Associated item', 'Associated items', 0)." (".__('Rear').")"."</th>";
+         echo "<th>".Socket::getTypeName(1)." (".__('Rear').")"."</th>";
+         echo "</tr>";
+
+         while ($data = $iterator->next()) {
+            $item = getItemForItemtype($data['type']);
+            $item->getFromDB($data['id']);
```

Comment:
Sometimes plugins are not correctly uninstalled, or items relations are not correctly cleaned.
```suggestion
            if (!($item = getItemForItemtype($data['type']) || !$item->getFromDB($data['id'])) {
               trigger_error(sprintf('Unable to load item %s (%s).', $data['type'], $data['id']), E_USER_WARNING);
               continue;
            }
```